src/simsoptpp/curveplanarfourier.cpp

```diff
@@ -0,0 +1,1633 @@
+#include "curveplanarfourier.h"
+
+
+template<class Array>
+void CurvePlanarFourier<Array>::gamma_impl(Array& data, Array& quadpoints) {
+    int numquadpoints = quadpoints.size();
+    data *= 0;
+
+    /* Converts q dofs to unit quaternion */
+    Array q_norm = xt::zeros<double>({4});
+    double s = q[0] * q[0] + q[1] * q[1] + q[2] * q[2] + q[3] * q[3];
+    if(s != 0) {
+        for (int i = 0; i < 4; ++i)
+            q_norm[i] = q[i] / std::sqrt(s); 
+    }
+    else {
+        q_norm[0] = 1;
+    }
+
+
+    for (int k = 0; k < numquadpoints; ++k) {
+        double phi = 2 * M_PI * quadpoints[k];
+        for (int i = 0; i < order+1; ++i) {
+            data(k, 0) += rc[i] * cos(i*phi) * cos(phi);
+            data(k, 1) += rc[i] * cos(i*phi) * sin(phi);
+        }
+    }
+    for (int k = 0; k < numquadpoints; ++k) {
+        double phi = 2 * M_PI * quadpoints[k];
+        for (int i = 1; i < order+1; ++i) {
+            data(k, 0) += rs[i-1] * sin(i*phi) * cos(phi);
+            data(k, 1) += rs[i-1] * sin(i*phi) * sin(phi);
+        }
+    }
+    for (int m = 0; m < numquadpoints; ++m) {
+        double i;
+        double j;
+        double k;
+        i = data(m, 0);
+        j = data(m, 1);
+        k = data(m, 2);
+
+        data(m, 0) = (i - 2 * (q_norm[2] * q_norm[2] + q_norm[3] * q_norm[3]) * i + 2 * (q_norm[1] * q_norm[2] - q_norm[3] * q_norm[0]) * j + 2 * (q_norm[1] * q_norm[3] + q_norm[2] * q_norm[0]) * k) + center[0];
```

Comment:
lines 43, 44, 45 are opaque to me.  Do they complete the solid body rotation? can you comment the code here to explain, and maybe give a reference?